api/src/OpenTelemetry/Internal/Trace/Types.hs

```diff
@@ -306,6 +306,17 @@ instance Show Span where
   showsPrec d (Dropped ctx) = showParen (d > 10) $ showString "Dropped " . showsPrec 11 ctx
 
 
+data FrozenOrDropped = SpanFrozen | SpanDropped deriving (Show, Eq)
+
+
+-- | Extracts the values from a @Span@ if it is still mutable. Returns @Nothing@ if the @Span@ is frozen or dropped.
```

Comment:
```suggestion
-- | Extracts the values from a @Span@ if it is still mutable. Returns @Either@
-- of @SpanFrozen@ if the @Span@ is frozen or @SpanDropped@ if it is dropped.
```
ymmv on wording, but it looks like the `@Nothing@` doc might be a holdover from a prior revision